### PR TITLE
[NCL-6205] make pnc-parallelism-threshold configurable

### PR DESCRIPTION
**Pull request by michalovjan**

### CODE REVIEW

**Comment by dwalluck** [comment]
There was already `pnc-partition-size` which was indirectly meant to control the number of threads. I see it is now used with:

```
configurationBuilder.pageSize(config.getPncPartitionSize());
```

which I am not sure is how it was intended and I'd prefer the name `pnc-num-threads` to match the `koji` option.

File: README.md [hunk 1]
```diff
@@ -251,6 +252,9 @@ The `use-checksums-file` and `use-builds-file` options specify whether
 to load any existing `checksums.json` or `builds.json` file,
 respectively. These files are always written, but not loaded by default.
 
+The `pnc-parallelism-threshold` signifies how many threads will be used
```
Comment (dwalluck) [inline README.md:255]: This is also sorted by name to match the order above.

File: core/src/main/java/org/jboss/pnc/build/finder/core/BuildConfig.java [hunk 2]
```diff
@@ -374,6 +377,18 @@ public void setUseChecksumsFile(Boolean useChecksumsFile) {
         this.useChecksumsFile = useChecksumsFile;
     }
 
+    public Long getPncParallelismThreshold() {
```
Comment (dwalluck) [inline core/src/main/java/org/jboss/pnc/build/finder/core/BuildConfig.java:380]: This is also sorted by name.

File: core/src/main/java/org/jboss/pnc/build/finder/core/ConfigDefaults.java [hunk 3]
```diff
@@ -68,6 +68,7 @@
     public static final URL PNC_URL = null;
     public static final Boolean USE_BUILDS_FILE = Boolean.FALSE;
     public static final Boolean USE_CHECKSUMS_FILE = Boolean.FALSE;
+    public static final Long PNC_PARALLELISM_THRESHOLD = 10L;
```
Comment (dwalluck) [inline core/src/main/java/org/jboss/pnc/build/finder/core/ConfigDefaults.java:71]: This is also sorted by name.

File: core/src/main/java/org/jboss/pnc/build/finder/core/BuildConfig.java [hunk 4]
```diff
@@ -96,6 +96,9 @@
     @JsonAlias("use-checksums-file")
     private Boolean useChecksumsFile;
 
+    @JsonAlias("pnc-parallelism-threshold")
+    private Long pncParallelismThreshold;
+
```
Comment (dwalluck) [inline core/src/main/java/org/jboss/pnc/build/finder/core/BuildConfig.java:101]: You also need to add the option to `cli/src/main/java/org/jboss/pnc/build/finder/cli/Main.java`.

Comment (michalovjan) [reply core/src/main/java/org/jboss/pnc/build/finder/core/BuildConfig.java:101]: sure thing

**Comment by michalovjan** [comment]
@dwalluck Sure, I am completely fine with the change to `pnc-num-threads`. OTOH, I wouldn't change the `pnc-partition-size` to `pnc-page-size` as this would break existing configurations which is not necessary and the current naming is apt.

**Comment by dwalluck** [comment]
> @dwalluck Sure, I am completely fine with the change to `pnc-num-threads`. OTOH, I wouldn't change the `pnc-partition-size` to `pnc-page-size` as this would break existing configurations which is not necessary and the current naming is apt.

To me it is confusing. I guess one possible thing is add `@JsonAlias`, but I guess it's too late now.

@michalovjan One last thing, can you add a line to `core/src/test/java/org/jboss/pnc/build/finder/core/BuildConfigTest#verifyDefaults` to verify the new config option?

**Comment by michalovjan** [comment]
@dwalluck Done :)

**Review by dwalluck — APPROVED** [review]

**Comment by michalovjan** [comment]
@dwalluck Is it good to merge?

**Pull request merged**